File: CPP/PROGRAM/Mouse.cpp
#pragma once
#include <iostream>
#include <string>
#include "Peripheral.cpp"

using namespace std;

class Mouse : public Peripheral{
    //atribut
    private:
        int dpi;
        int jumlahTombol;

    public:
        //konstruktor
        Mouse(){
            this->dpi = 0;
            this->jumlahTombol = 0;
        }

        Mouse(int dpi, int jumlahTombol, string tipePeripheral, string interface, string merk, string nama) : Peripheral(tipePeripheral, interface, merk, nama){
            this->dpi = dpi;
            this->jumlahTombol = jumlahTombol;
        }

        //getter dan setter
        void setDPI(int dpi){
            this->dpi = dpi;
        }

        void setJumlahTombol(int jumlahTombol){
            this->jumlahTombol = jumlahTombol;
        }        

        int getDPI(){
            return this->dpi;
        }

        int getJumlahTombol(){
            return this->jumlahTombol;
        }

        ~Mouse(){

        }
};

File: PYTHON/PROGRAM/Mouse.py
from Peripheral import Peripheral

class Mouse(Peripheral):
    #konstruktor
    def __init__(self, tipePeripheral = "", interface = "", merk = "", nama = "", dpi = 0, jumlahTombol = 0):
        super().__init__(merk, nama, tipePeripheral, interface)
        self.__dpi = dpi
        self.__jumlahTombol = jumlahTombol

    #getter dan setter
    def getDpi(self):
        return self.__dpi
    
    def setDpi(self, dpi):
        self.__dpi = dpi

    def getJumlahTombol(self):
        return self.__jumlahTombol
    
    def setJumlahTombol(self, jumlahTombol):
        self.__jumlahTombol = jumlahTombol